src/coreclr/vm/assembly.cpp

```diff
@@ -1392,7 +1449,11 @@ INT32 Assembly::ExecuteMainMethod(PTRARRAYREF *stringArgs, BOOL waitForOtherThre
 
             hr = RunMain(pMeth, 1, &iRetVal, stringArgs);
 
+// Keep the thread attached, because in the browser we keep the runtime alive.
```

Comment:
`CleanUpForManagedThreadInNative` is not detaching anything. It is doing Apple-specific per-workitem cleanup. The same cleanup is done on thread pool after each workitem executes (see `DispatchItemWithAutoreleasePool`).

This comment should be deleted. The BROWSER ifdef is unnecessary - it is ifdefing a call to a empty function that the compiler should have no problem optimizing out - so it can be deleted too.